src/map/utils/battleutils.cpp

```diff
@@ -1802,27 +1803,49 @@ namespace battleutils
                 return 0;
             }
         }
-        else if (PDefender->objtype == TYPE_MOB && PDefender->GetMJob() == JOB_PLD)
+        else if (PDefender->objtype != TYPE_PC)
         {
-            CMobEntity* PMob = (CMobEntity*)PDefender;
-
-            if (PMob->m_EcoSystem != ECOSYSTEM::UNDEAD && PMob->m_EcoSystem != ECOSYSTEM::BEASTMAN)
+            CMobEntity* PEntity = (CMobEntity*)PDefender;
+            if (PEntity->getMobMod(MOBMOD_CAN_SHIELD_BLOCK) > 0)
             {
-                return 0;
-            }
-        }
-        else if (PDefender->objtype == TYPE_PET && static_cast<CPetEntity*>(PDefender)->getPetType() == PET_TYPE::AUTOMATON && PDefender->GetMJob() == JOB_PLD)
-        {
-            skillModifier = (int8)((PDefender->GetSkill(SKILL_AUTOMATON_MELEE) - attackSkill) * 0.215f);
-            base          = PDefender->getMod(Mod::SHIELDBLOCKRATE);
+                base = PDefender->getMod(Mod::SHIELDBLOCKRATE);
+                if (PDefender->objtype == TYPE_PET)
+                {
+                    skillModifier = (int8)((PDefender->GetSkill(SKILL_AUTOMATON_MELEE) - attackSkill) * 0.215f);
+                    if (base <= 0)
+                    {
+                        return 0;
+                    }
+                    else
+                    {
+                        return base + skillModifier;
+                    }
+                }
+                else
+                {
+                    // Check for Reprisal and adjust skill and block rate bonus multiplier
+                    if (PDefender->StatusEffectContainer->HasStatusEffect(EFFECT_REPRISAL))
```

Comment:
We need a skill modifier check --
      
```
            // TODO: check trust type for ilvl > 99 when implemented
            blockSkill = GetMaxSkill(SKILLTYPE::SKILL_SHIELD, PDefender->GetMJob(), PDefender->GetMLevel() > 99 ? 99 : PDefender->GetMLevel());
```
 need the multiplier for block skill again from reprisal + the other block skill modifiers in from type PC

Reply:
That's the bit I was missing before Teo kept trying to tell me to put in but I misunderstood him. 😭

Reply:
> We need a skill modifier check --
> 
>       
> 
> ```
> 
>             // TODO: check trust type for ilvl > 99 when implemented
> 
>             blockSkill = GetMaxSkill(SKILLTYPE::SKILL_SHIELD, PDefender->GetMJob(), PDefender->GetMLevel() > 99 ? 99 : PDefender->GetMLevel());
> 
> ```
> 
>  need the multiplier for block skill again from reprisal + the other block skill modifiers in from type PC
> 
> 

- Block flat values prior to skill check
    - 30 flat (mod)
    - 45 +15% reprisal
    - 60 +30% palisade 
![image](https://user-images.githubusercontent.com/99501957/198698444-0e26fd7a-08cb-4231-a091-f7e04dbd8659.jpeg)

- Block with max skill check added
    - 35 flat (mod) +5% 
    - 74 +39% Reprisal
    - 65 +30% palisade 
![image](https://user-images.githubusercontent.com/99501957/198699453-9340d862-961e-40ae-a516-1261a751cddc.jpeg)